### ISSUE TITLE
Breakage after upgrading j4ts

### ISSUE THREAD

**Issue by mb720** [cincheo/jsweet#158]
Hi,

I tried to compile my JSweet project whereupon it updated some libraries:
```
Download http://repository.jsweet.org/artifactory/libs-snapshot-local/org/jsweet/candies/j4ts/0.2.0-SNAPSHOT/j4ts-0.2.0-20160831.153627-3.pom
Download http://repository.jsweet.org/artifactory/libs-snapshot-local/org/jsweet/candies/sprintf-js/0.0.0-SNAPSHOT/sprintf-js-0.0.0-20160726.052418-37.pom
Download http://repository.jsweet.org/artifactory/libs-snapshot-local/org/jsweet/candies/j4ts/0.2.0-SNAPSHOT/j4ts-0.2.0-20160831.153627-3.jar
Download http://repository.jsweet.org/artifactory/libs-snapshot-local/org/jsweet/candies/sprintf-js/0.0.0-SNAPSHOT/sprintf-js-0.0.0-20160726.052418-37.jar
```

During compilation I got this suspicious looking warning (the suspicious thing about it is that the candy version is null):

`candy j4ts:0.2.0-SNAPSHOT was generated for a different version of the transpiler (current:1.2.0, candy:null)`

Then began the errors:

```
error retreiving annotation class jsweet.lang.Root
cannot read mixins info for interface def.es6_promise.package-info in C:\Users\me\.gradle\caches\modules-2\files-2.1\org.jsweet.candies\es6-promise\0.0.0-SNAPSHOT\25fcb03c7a2614ef47344eb3a3a56322dfb71b57\es6-promise-0.0.0-SNAPSHOT.jar
error retreiving annotation class jsweet.lang.Root
cannot read mixins info for interface def.es6_promise.es6_promise.package-info in C:\Users\me\.gradle\caches\modules-2\files-2.1\org.jsweet.candies\es6-promise\0.0.0-SNAPSHOT\25fcb03c7a2614ef47344eb3a3a56322dfb71b57\es6-promise-0.0.0-SNAPSHOT.jar
error retreiving annotation class jsweet.lang.Root
cannot read mixins info for interface def.jquery.package-info in C:\Users\me\.gradle\caches\modules-2\files-2.1\org.jsweet.candies\jquery\1.10.0-SNAPSHOT\f0ffebea6c24246f5daec8eb8dd42eb9fc9ea97a\jquery-1.10.0-SNAPSHOT.jar
error retreiving annotation class jsweet.lang.Root
cannot read mixins info for interface def.marked.package-info in C:\Users\me\.gradle\caches\modules-2\files-2.1\org.jsweet.candies\marked\0.0.0-SNAPSHOT\83cbe3cfe2924d24c8256e7d363f5d4abcc6cf98\marked-0.0.0-SNAPSHOT.jar
error retreiving annotation class jsweet.lang.Root
cannot read mixins info for interface def.sprintf_js.package-info in C:\Users\me\.gradle\caches\mes-2\files-2.1\org.jsweet.candies\sprintf-js\0.0.0-SNAPSHOT\6b2e6a2c64e3ad7477c2557ae5a55177bc5df12b\sprintf-js-0.0.0-SNAPSHOT.jar
error retreiving annotation class jsweet.lang.Root
cannot read mixins info for interface def.sprintf_js.sprintf_js.package-info in C:\Users\me\.gradle\caches\modules-2\files-2.1\org.jsweet.candies\sprintf-js\0.0.0-SNAPSHOT\6b2e6a2c64e3ad7477c2557ae5a55177bc5df12b\sprintf-js-0.0.0-SNAPSHOT.jar
invalid package hierarchy: type 'Object' is declared in a parent of @Root package 'my.package' at C:\Users\me\path\to\my\Class.java(1)
```
Any help is appreciated.

Best,
Matthias

**Comment by renaudpawlak** [comment]
It looks like I have broken something by changing the way J4TS is packaged... I will fix ASAP... sorry for the inconvenience.

**Comment by renaudpawlak** [comment]
I have redeployed. It should work better now.
I still have a weird error... 

```
2016-09-04 20:41:04.004 ERROR CandiesMerger:147 - error retreiving annotation class jsweet.lang.Root
java.lang.SecurityException: Prohibited package name: java.lang.annotation
```

It does not seem to have much impact so far... so I will investigate later.
Thanks!

**Comment by renaudpawlak** [comment]
I have also fixed the security exception in the latest JSweet snapshot... I have redeployed J4TS with ``java.awt.geom`` support... :D Please tell me if it works good for you.

**Comment by mb720** [comment]
Thanks for the quick response and adding `java.awt.geom`, Renaud.

I get this error now, but I'm not sure if the problem is on my end:

`invalid package hierarchy: type 'Object' is declared in a parent of @Root package 'eu.my.packagename' at C:\Users\me\path\to\a\JavaClass.java(1)`

**Comment by renaudpawlak** [comment]
Umm... nothing in JSweet has changed in that direction I think.
Have you used the `@Root` annotation in a `package-info.java` file?
As far as I remember (and as the error messages says ;)), this error occurs when you define a `@Root` package and have some defined class in a parent package (`@Root` packages are the root of the package hierarchy, so it is not allowed to define anything out... there are other restrictions with multiple `@Root`).
In short, first check that you don't have any `package-info.java` files containing a `@Root` annotation.

**Comment by mb720** [comment]
I actually do have a `package-info.java` with a `@Root` annotation, but that hasn't caused troubles before.

The curious thing is that the error complains about `Object` (I take it that's `jsweet.lang.Object`) which is of course outside the package I've annotated with `@Root`.

**Comment by renaudpawlak** [comment]
Ok. This is interesting. There is an `Object` class defined in the default (empty) package in J4TS. It  seems very likely that there is some side effect here... So it's clearly a bug to me. I will fix ASAP. Thanks for detecting it!
